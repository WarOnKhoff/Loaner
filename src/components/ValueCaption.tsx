import React, { FC, ReactElement } from "react"

interface IProps {
  caption: string
  value: number
  postfix?: string
  testId?: string
}
const ValueCaption: FC<IProps> = ({
  caption,
  value,
  postfix,
  testId
}: IProps): ReactElement => (
  <div className="summary_group">
    <div className="summary_group_text" data-testid="capture_text">
      {caption}
    </div>
    <div className="summary_group_flex">
      <div className="summary_group_value" data-testid={testId}>
        {value}
      </div>
      <div className="summary_group_postfix" data-testid="capture_postfix">
        {postfix}
      </div>
    </div>
  </div>
)

export default React.memo(ValueCaption)
