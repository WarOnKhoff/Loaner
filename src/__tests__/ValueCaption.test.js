import React from "react"
import { render, fireEvent } from "@testing-library/react"
import ValueCaption from "../components/ValueCaption"

const setup = props => {
  return render(<ValueCaption {...props} />)
}
const mocks = {
  caption: "Test caption",
  value: "Test Value",
  postfix: "Test postfix",
  testId: "caption_test"
}
describe("Components: ValueCaption", () => {
  test("Renders correctly", () => {
    const { container } = setup()
    expect(container).toMatchSnapshot()
  })
  test("Should render proper caption from props", () => {
    const { getByTestId } = setup(mocks)
    expect(getByTestId("capture_text").innerHTML).toBe(mocks.caption)
  })
  test("Should render proper value from props", () => {
    const { getByTestId } = setup(mocks)
    expect(getByTestId(mocks.testId).innerHTML).toBe(mocks.value)
  })
  test("Should render proper postfix from props", () => {
    const { getByTestId } = setup(mocks)
    expect(getByTestId("capture_postfix").innerHTML).toBe(mocks.postfix)
  })
})
