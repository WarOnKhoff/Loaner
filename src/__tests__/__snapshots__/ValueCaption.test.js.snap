// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: ValueCaption Renders correctly 1`] = `
<div>
  <div
    class="summary_group"
  >
    <div
      class="summary_group_text"
      data-testid="capture_text"
    />
    <div
      class="summary_group_flex"
    >
      <div
        class="summary_group_value"
      />
      <div
        class="summary_group_postfix"
        data-testid="capture_postfix"
      />
    </div>
  </div>
</div>
`;
